src/net/server/handlers/create_query.rs

```diff
@@ -0,0 +1,264 @@
+            *buffered_size -= size_in_bytes;
+            out_bytes
+        } else {
+            // First buffer is too small, so we need to take across buffers.
+            // Will require a memcopy of u8's across buffers in order to create 1 contiguous buffer
+            // to return.
+            // Removes expended buffers after taking
+
+            let mut out_bytes = Vec::with_capacity(size_in_bytes as usize);
+            // this must loop through the bytes buffers because we don't know how many buffers will
+            // be needed to fulfill `size_in_bytes`. e.g. if every buffer had length 1, we'd need to
+            // visit `size_in_bytes` buffers in order to fill `out_bytes`
+            loop {
+                let remaining_bytes = out_bytes.capacity() - out_bytes.len();
+                if remaining_bytes == 0 {
+                    break;
+                }
+                // current `buffer` has more bytes than needed
+                if buffered[0].len() > remaining_bytes {
+                    let remaining = buffered[0].split_to(remaining_bytes);
+                    out_bytes.extend_from_slice(&remaining);
+                } else {
+                    // current `buffer` has <= bytes needed, remove and append to out_bytes
+                    out_bytes.extend_from_slice(&buffered.pop_front().unwrap());
+                }
+            }
+            // reduce size of total buffers accordingly
+            *buffered_size -= size_in_bytes;
+
+            Bytes::from(out_bytes)
+        }
+    }
+}
+
+impl Stream for ByteArrStream {
+    type Item = Result<Bytes, std::io::Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let mut this = self.as_mut().project();
+        loop {
+            // if we currently have enough bytes, return them
+            if *this.buffered_size >= *this.size_in_bytes {
+                return Poll::Ready(Some(Ok(Self::take_bytes(
+                    *this.size_in_bytes,
+                    this.buffered,
+                    this.buffered_size,
+                ))));
+            }
+            // if we need more bytes, poll the body
+            match ready!(this.body.as_mut().poll_next(cx)) {
+                // if body is expended, but we have some bytes leftover, error
+                None if *this.buffered_size > 0 => {
+                    return Poll::Ready(Some(Err(std::io::Error::new::<BoxError>(
+                        std::io::ErrorKind::WriteZero,
+                        format!(
+                            "expected body to align on size {}, but has insufficient bytes {}",
+                            *this.size_in_bytes, *this.buffered_size
+                        )
+                        .into(),
+                    ))));
+                }
+                // if body is finished, this stream is finished
+                None => return Poll::Ready(None),
```

Comment:
you may have some leftover bytes in the buffer here, so it is potentially an error too

Reply:
is this not handled in the case statement right above? 

> None if *this.buffered_size > 0 => {